packages/state-transition/src/util/blindedBlock.ts

```diff
@@ -78,9 +84,14 @@ export function signedBeaconBlockToBlinded(
 }
 
 export function signedBlindedBlockToFull(
+  fork: ForkName,
   signedBlindedBlock: SignedBlindedBeaconBlock,
   executionPayload: ExecutionPayload | null
 ): SignedBeaconBlock {
+  if (isForkPostBellatrix(fork) && executionPayload === null) {
+    throw Error("Missing executionPayload for post-bellatrix blinded block");
```

Comment:
this error could be misleading, maybe

```suggestion
    throw Error("Missing executionPayload to construct post-bellatrix full block");
```